src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -42,345 +128,186 @@ object WdlNamespace {
-  val combinedTerminalMap = ((namespaces map {x => x.terminalMap}) ++ Seq(terminalMap)) reduce (_ ++ _)
-  val wdlSyntaxErrorFormatter = new WdlSyntaxErrorFormatter(combinedTerminalMap)
+  def load(wdlSource: WdlSource, resource: String): WdlNamespace = load(wdlSource, resource, localImportResolver, None)
 
-  /**
-   * All imported `task` definitions for `import` statements without a namespace (e.g. no `as` clause)
-   * These tasks are considered to be in this current workspace
-   */
-  val importedTaskAsts: Seq[Ast] = namespaces flatMap { b =>
-    b.namespace match {
-      case None => b.taskAsts
-      case _ => Seq.empty[Ast]
-    }
-  }
-  val importedTasks: Seq[Task] = namespaces flatMap { b =>
-    b.namespace match {
-      case None => b.tasks
-      case _ => Seq.empty[Task]
-    }
+  def load(wdlSource: WdlSource, resource: String, importResolver: ImportResolver): WdlNamespace = {
+    load(wdlSource, resource, importResolver, None)
   }
 
-  /**
-   * All `task` definitions defined in the WDL file (i.e. not imported)
-   */
-  val localTaskAsts = ast.findAsts(AstNodeName.Task)
-  val localTasks = localTaskAsts.map(processTask)
-
-  /**
-   * All `task` definitions, including local and imported ones
-   */
-  val taskAsts = localTaskAsts ++ importedTaskAsts
-  val tasks = localTasks ++ importedTasks
-
-  /**
-   * All imported `Workflow`s
-   */
-  val importedWorkflows: Seq[Workflow] = namespaces flatMap { b =>
-    b.namespace match {
-      case None => b.workflows
-      case _ => Seq.empty[Workflow]
-    }
+  private def load(wdlSource: WdlSource, resource: String, importResolver: ImportResolver,
+                   importedAs: Option[String]): WdlNamespace = {
+    WdlNamespace(AstTools.getAst(wdlSource, resource), wdlSource, importResolver, importedAs)
   }
 
   /**
-   * All `Workflow`s defined in the WDL file (i.e. not imported)
-   */
-  val localWorkflows = ast.findAsts(AstNodeName.Workflow).map(processWorkflow)
-
-  /**
-   * All `Workflow` definitions, including local and imported ones
+   * Validates the following things about the AST:
+   *
+   * 1) Tasks do not have duplicate inputs
+   * 2) Tasks in this namespace have unique names
+   * 3) Tasks and namespaces don't have overlapping names
    */
-  val workflows = importedWorkflows ++ localWorkflows
-
-  lazy val workflow = workflows.head
-
-  lazy val calls = workflow.calls
+  def apply(ast: Ast, source: WdlSource, importResolver: ImportResolver, namespace: Option[String]): WdlNamespace = {
+    /**
+     * All `import` statement strings at the top of the document
+     */
+    val imports = ast.getAttribute("imports").asInstanceOf[AstList].asScala map {x => Import(x)}
 
-  /* All `Task`s and `Workflow`s */
-  val executables = workflows ++ tasks
+    /* WdlBinding objects for each import statement */
+    val namespaces: Seq[WdlNamespace] = {for {
+      i <- imports
+      source = importResolver(i.uri) if source.length > 0
```

Comment:
Existential question, is an empty WDL valid? It's not runnable but it could be valid.